bolt/src/main/java/com/slack/api/bolt/middleware/builtin/MultiTeamsAuthorization.java

```diff
@@ -31,6 +36,16 @@
     private final AppConfig config;
     private final InstallationService installationService;
 
+    @Data
+    @AllArgsConstructor
+    static class CachedAuthTestResponse {
+        private AuthTestResponse response;
+        private long cachedMillis;
+    }
+
+    // token -> auth.test response
+    private final ConcurrentMap<String, CachedAuthTestResponse> tokenToAuthTestCache = new ConcurrentHashMap<>();
```

Comment:
We may want to clean up very old cache data to avoid excessive memory usage in extreme cases. As I don't want to add any dependencies only for this, we can implement a simple mechanism on our own.

Reply:
done